site/blog/top-5-open-source-ai-red-teaming-tools-2025.md

```
@@ -0,0 +1,235 @@
+
+## Top open Source AI red teaming tools (2025)
+
+Bear in mind that we're looking to run red teams, not just evals. I've omitted tools that only run evals; while evals can be used as a part of a red teaming workflow (and we certainly utilise evals in our AI red teaming process), that is only part of the process in highlighting security risks from AI responses.
+
+### 1. Promptfoo
+
+You're on the Promptfoo blog, so let's get this one out of the way.
+
+**Overview**: A dev-first framework for evaluating and [red teaming LLMs](https://www.promptfoo.dev/docs/red-team/agents/) that works by automatically generating attack prompts deploying them using various strategies.
+    
+**Key Features**
```

markdownlint-cli2 (0.17.2)

Emphasis used instead of a heading

(MD036, no-emphasis-as-heading)